var/spack/repos/builtin/packages/minizip/package.py

```diff
@@ -37,6 +38,15 @@ def build_minizip_binary(self):
         with working_dir(self.configure_directory):
             make()
 
+    # Building a shared lib against a static one is not portable
+    # so we link against the PIC independent lib
+    def build(self, spec, prefix):
+        with working_dir(self.configure_directory):
+            makefile = FileFilter('Makefile')
+            makefile.filter('libz.a', 'libz.so')
```

Comment:
This seems a better fit for a `patch` method

Reply:
Oh I just found out it was the "patch('static.patch')" that makes the build process fail on Ubuntu.
Could we apply the patch only when='%apple-clang@12:' ?
Probably @adamjstewart could provide some advise ?
Not applying the patch is successfull on Ubuntu-18 x86_64 gcc and aocc + arm gcc and clang.
I do not have a working xcode env on mac to test any further